client/src/screens/ProductScreen/RenderProductScreen.js

```diff
@@ -10,12 +13,59 @@ import product3 from "../../assets/images/product3.jpg";
 import product4 from "../../assets/images/product4.jpg";
 import "./productScreen.css";
 
-const RenderProductScreen = () => {
-  const addToCartHandler = (id) => {
-    // need to work on this
-    console.log("need to figure this out");
+const initialState = {
+  product_name: "",
+  product_price: "",
+  product_size: "",
+};
+
+const RenderProductScreen = ({ match }) => {
+  const [quantity, setQuantity] = useState(0);
+  const [sizes, setSizes] = useState([]);
+  const [productToAdd, setProductToAdd] = useState(initialState);
+
+  const productDetails = useSelector((state) => state.productDetails);
+  const {
+    loading: loadingDetails,
+    error: errorDetails,
+    product,
+  } = productDetails;
+
+  const dispatch = useDispatch();
+
+  const { id } = useParams();
+
+  const userLogin = useSelector((state) => state.userLogin);
+  const { userInfo } = userLogin;
+
+  console.log(userInfo);
+  console.log("product", product);
+
+  const productList = useSelector((state) => state.productList);
+  const { loading, error, products } = productList;
+
+  useEffect(() => {
+    dispatch(listProductDetails(id));
+  }, [dispatch]);
+
+  const addToCartHandler = (e) => {
+    e.preventDefault();
+    const newObj = {
+      ...initialState,
+      user_id: userInfo.user_id,
+      product_price: product.price,
+      product_name: product.name,
+      product_img: product.images[0].img_url,
+      product_size: product.sizes[0].size,
```

Comment:
```size```  should be coming from user input right? not from the actual data
for example, a user can choose ( XXL, xl, large, medium, small) user needs to pick any of those